src/components/Layout/index.tsx

```diff
@@ -16,9 +19,21 @@ const messages = defineMessages({
 const Layout: React.FC = ({ children }) => {
   const [isSidebarOpen, setSidebarOpen] = useState(false);
   const [isScrolled, setIsScrolled] = useState(false);
-  const { hasPermission } = useUser();
+  const { user, hasPermission } = useUser();
   const router = useRouter();
   const intl = useIntl();
+  const { currentSettings } = useSettings();
+  const { setLocale } = useLocale();
+
+  useEffect(() => {
+    if (setLocale) {
+      setLocale(
+        (user?.settings?.locale
+          ? user.settings.locale
+          : currentSettings.locale) as AvailableLocales
+      );
+    }
+  }, [setLocale, currentSettings.locale, user]);
```

Comment:
This also means though that after the setup is completed for the first time the language will reset back to English.

Reply:
Yes, but if the user refreshes the page after completing setup, it will also reset back to English. 🤔

Perhaps at the end of setup, we should save the selected locale to settings?